core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -400,7 +398,9 @@ private[kafka] class Processor(val id: Int,
 
   private val newConnections = new ConcurrentLinkedQueue[SocketChannel]()
   private val inflightResponses = mutable.Map[String, RequestChannel.Response]()
-  private val metricTags = Map("networkProcessor" -> id.toString).asJava
+  val metricTags = Map(
+    "protocol" -> securityProtocol.name, "listener" -> listenerName.value, "networkProcessor" -> id.toString
+    ).asJava
 
   newGauge("IdlePercent",
```

Comment:
You didn't do the same for `NetworkProcessorAvgIdlePercent` though. My preference would be to leave the Yammer metrics alone and just add the tags to the underlying SelectorMetrics, but worth waiting for @junrao's opinion.